code/modules/antagonists/changeling/powers/fakedeath.dm

```diff
@@ -51,14 +50,21 @@
 /// Can be called mid-revival if the process is being cancelled
 /datum/action/changeling/fakedeath/proc/disable_revive(mob/living/changeling)
 	if(revive_ready)
-		chemical_cost = 15
-		revive_ready = FALSE
-		build_all_button_icons(UPDATE_BUTTON_NAME|UPDATE_BUTTON_ICON)
+		addtimer(CALLBACK(src, PROC_REF(reset_chemical_cost)), 2 DECISECONDS, TIMER_UNIQUE)
+		//a timer is used here because this needs to be done after the upstream logic
+		//of try_to_sting()
```

Comment:
Doesn't this mean you can spam click the button after reviving (in the 2 decisecond window, to go right back into revival stasis for free? (Not that I think it is extremely useful per se but still?)

Reply:
Or actually thinking about it more it's not going back into revival stasis, you can call revive() multiple times per stasis

Reply:
Oh, I didn't think of that, I'll have to take another look.

Reply:
you should really just use the `COOLDOWN_DECLARE` system or something